File: android_library/mobiledownloadlibrary/src/main/java/com/mediamonks/mobiledownload/model/ProgressModel.java
package com.mediamonks.mobiledownload.model;

/**
 * Created by raphael on 15/04/16.
 */
public class ProgressModel {

//        DOWNLOADING = 0,
//        SUSPENDED = 1

    public DownloadItem mFile;

    public int mProgress;

    public int mProgressType = 0;

    public int mGroupPosition = 0;

    public int mGroupSize = 1;

    public ProgressModel() {
        // no-args constructor
    }

    public static ProgressModel empty() {
        ProgressModel progressModel = new ProgressModel();
        progressModel.mProgress = 0;
        progressModel.mFile = null;
        progressModel.mProgressType = 0;
        progressModel.mGroupPosition = 0;
        progressModel.mGroupSize = 1;
        return progressModel;
    }
}

File: Assets/Plugins/MobileDownloadPlugin/Scripts/ProgressModel.cs
using UnityEngine;
using System.Collections;

public class ProgressModel {

	public enum ProgressType{
		DOWNLOADING = 0,
		SUSPENDED = 1
	};

	public FileModel mFile;

	public ProgressType mProgressType;

	public int mProgress;

	public int mGroupSize = 1;//optional fields

	public int mGroupPosition = 0;//optinal fields

	public ProgressModel(){}

	public ProgressModel(string messageJson){
		JSONObject j = new JSONObject(messageJson);
		mProgressType = (ProgressType) j.GetField ("mProgressType").i;
		mProgress = (int) j.GetField ("mProgress").i;
		mFile = new FileModel (j.GetField ("mFile").ToString ());
		if (j.HasField ("mGroupSize")) {
			mGroupSize = (int) j.GetField ("mGroupSize").i;
		}
		if (j.HasField ("mGroupPosition")) {
			mGroupPosition = (int) j.GetField ("mGroupPosition").i;
		}
	}
}
